### DISCUSSION TITLE
F# REPL improvements? look at C# repl global dotnet tools

### DISCUSSION THREAD

**Discussion by jkone27** [dotnet/fsharp#15469] [Ideas]
Seems C# repl is quite pleasant to use from terminal with nice colorization, IntelliSense and colors... maybe interesting if it could be possible to reuse some of this stuff or "PLUG IT" to F# dotnet fsi REPL ?

![image](https://github.com/dotnet/fsharp/assets/10419217/4750ad92-34f4-42c8-989b-6c4fa867c050)

https://www.youtube.com/shorts/CJpjjPlI0ss

https://github.com/waf/CSharpRepl

https://github.com/jonsequitur/dotnet-repl ---> also nice, thought C# repl seems to have more features...

**Comment by baronfel** [comment]
One note is that tools bundled into the .NET SDK must be source-buildable - meaning they and all of their dependencies must be build able via the mechanism described in the dotnet/source-build repo. This makes it simple for anyone to build a distribution of the SDK and tools.

Practically, what it means is that there's a pretty high bar to bring in dependencies, and all of these tools use many dependencies to make the actual terminal interactions easier/prettier/more user friendly.
